x/wasm-storage/keeper/abci.go

```diff
@@ -32,3 +62,50 @@ func (k Keeper) ProcessExpiredWasms(ctx sdk.Context) error {
 	}
 	return nil
 }
+
+func (k Keeper) ExecuteTally(ctx sdk.Context) error {
+	// 1. Get contract address.
+	contractAddr, err := k.ProxyContractRegistry.Get(ctx)
+	if contractAddr == "" || errors.Is(err, collections.ErrNotFound) {
+		k.Logger(ctx).Debug("proxy contract address not registered")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// 2. Fetch tally-ready DRs.
+	// TODO: json marshal request struct?
+	queryRes, err := k.wasmViewKeeper.QuerySmart(ctx, sdk.MustAccAddressFromBech32(contractAddr), []byte(`{"get_data_requests_by_status":{"status": "tallying"}}`))
```

Comment:
You can select an amount to do at once via limit. Otherwise, this will grab ALL of them at once, which might be heavy. Would this impact block production speed? We might need to play around with the limit.

Reply:
good pointed - added to task list